### PR TITLE
build: Add support for a Dev Container

### PR DESCRIPTION
**Pull request by sitaowang1998**
# Description
Add basic setup for devconatiner. Devcontainer uses ubuntu focal as base, and install all the dependencies using install scripts in `./tools/scripts/lib_install/linux`.

To start the devcontainer in Clion, see [jetbrain document](https://www.jetbrains.com/help/clion/connect-to-devcontainer.html#create_dev_container_inside_ide).

# Validation performed
- [x] task:lint
- [x] Start devcontainer in Clion on local macOS.

## Summary by CodeRabbit

- **New Features**
	- Introduced a development container configuration for a consistent development environment.
	- Added scripts for automating the installation of development tools and libraries on Linux and macOS.

- **Bug Fixes**
	- Updated directory paths in task configurations to ensure correct library installation locations.

- **Chores**
	- Cleaned up temporary files in the Dockerfile to reduce image size.
	- Removed an outdated library installation script from the Linux directory.

### CODE REVIEW

**Comment by sitaowang1998** [comment]
Right now the container does not install `python3-venv` library, which means user needs to install it manually to run lint tasks. I'll add a commit to solve this later.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "build": {
+    "dockerfile": "Dockerfile",
+    "context": "..",
+  },
+}
```
Comment (kirkrodrigues) [inline .devcontainer/devcontainer.json:6]:
```suggestion
}

```

File: tools/scripts/lib_install/linux/install-dev.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  gcc-10 \
+  git \
+  libcurl4 \
+  libcurl4-openssl-dev \
+  libssl-dev \
+  make \
+  pkg-config \
+  python3 \
+  python3-pip \
+  python3-venv
```
Comment (kirkrodrigues) [inline tools/scripts/lib_install/linux/install-dev.sh:27]: Use four spaces for indenting code and scripts.

File: tools/scripts/lib_install/linux/install-dev.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# Add alternatives for cpp-10, gcc-10, and g++-10
+# NOTE: We use a low priority to avoid affecting the prioritization of any existing alternatives
+# on the user's system.
+update-alternatives --install /usr/bin/cc cc /usr/bin/gcc-10 0 \
+  --slave /usr/share/man/man1/cc.1.gz cc.1.gz /usr/share/man/man1/gcc.1.gz
+update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++-10 0 \
+  --slave /usr/share/man/man1/c++.1.gz c++.1.gz /usr/share/man/man1/g++.1.gz
+update-alternatives --install /lib/cpp cpp /usr/bin/cpp-10 0
```
Comment (kirkrodrigues) [inline tools/scripts/lib_install/linux/install-dev.sh:36]: These are no longer necessary, right?

File: tools/scripts/lib_install/macOS/install-lib.sh [hunk 4]
```diff
@@ -14,4 +14,4 @@ brew install \
 
 script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 lib_install_scripts_dir=$script_dir/..
-"${lib_install_scripts_dir}"/lib_install/mariadb-connector-cpp.sh 1.1.5
\ No newline at end of file
+"${lib_install_scripts_dir}"/mariadb-connector-cpp.sh 1.1.5
```
Comment (kirkrodrigues) [inline tools/scripts/lib_install/macOS/install-lib.sh:17]:
```suggestion
"${lib_install_scripts_dir}"/mariadb-connector-cpp.sh 1.1.5

```

File: tools/scripts/lib_install/linux/install-dev.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+apt-get update
+DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+  ca-certificates \
+  checkinstall \
+  cmake \
+  curl \
+  g++ \
+  g++-10 \
+  gcc \
+  gcc-10 \
```
Comment (kirkrodrigues) [inline tools/scripts/lib_install/linux/install-dev.sh:18]: g++-10 and gcc-10 still necessary?

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]
For the PR title, how about:

```
build: Add support for a Dev Container.
```

**Pull request merged**